src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -305,98 +297,84 @@ export function createModelForService(
-                     * it is client parameter
-                     * it does not has default value, or the default value is included in the global service apiVersion.
-                     */
-                    if (
-                        apiVersions.has(
-                            apiVersionInOperation.DefaultValue?.Value
-                        ) &&
-                        apiVersionInOperation.Kind ===
-                            InputOperationParameterKind.Client &&
-                        apiVersionInOperation.Location ===
-                            apiVersionParam.Location
-                    ) {
-                        op.Parameters[apiVersionIndex] = apiVersionParam;
-                    }
-                } else {
-                    op.Parameters.push(apiVersionParam);
+    for (const client of clients) {
+        for (const op of client.Operations) {
+            const apiVersionIndex = op.Parameters.findIndex(
+                (value) => value.IsApiVersion
+            );
+            if (apiVersionIndex !== -1) {
+                const apiVersionInOperation = op.Parameters[apiVersionIndex];
+                if (!apiVersionInOperation.DefaultValue?.Value) {
+                    apiVersionInOperation.DefaultValue =
+                        apiVersionParam.DefaultValue;
                 }
+                /**
+                 * replace to the global apiVerison parameter if the apiVersion defined in the operation is the same as the global service apiVersion parameter.
+                 * Three checkpoints:
+                 * the parameter is query parameter,
+                 * it is client parameter
+                 * it does not has default value, or the default value is included in the global service apiVersion.
+                 */
+                if (
+                    apiVersions.has(
+                        apiVersionInOperation.DefaultValue?.Value
+                    ) &&
+                    apiVersionInOperation.Kind ===
+                        InputOperationParameterKind.Client &&
+                    apiVersionInOperation.Location === apiVersionParam.Location
+                ) {
+                    op.Parameters[apiVersionIndex] = apiVersionParam;
+                }
+            } else {
+                op.Parameters.push(apiVersionParam);
             }
         }
-
-        const usages = getUsages(program, convenienceOperations);
-        setUsage(usages, modelMap);
-        setUsage(usages, enumMap);
-
-        const clientModel = {
-            Name: namespace,
-            Description: description,
-            ApiVersions: Array.from(apiVersions.values()),
-            Enums: Array.from(enumMap.values()),
-            Models: Array.from(modelMap.values()),
-            Clients: clients,
-            Auth: auth
-        } as CodeModel;
-        return clientModel;
-    } catch (err) {
-        if (err instanceof ErrorTypeFoundError) {
-            return;
```

Comment:
I removed this try catch because I found that outside this function, we never check if the code model returned in this function is undefined or not therefore we just throw the error out when something wrong happens instead of return undefined. This could also enable this function to return with a concrete type `CodeModel` which previously is `any`